bundler/spec/bundler/env_spec.rb

```diff
@@ -45,9 +45,15 @@
       it "prints user path" do
         skip "needs to use a valid HOME" if Gem.win_platform? && RUBY_VERSION < "2.6.0"
 
-        with_clear_paths("HOME", "/a/b/c") do
+        if Gem::VERSION >= "3.2.0.pre.1"
+          allow(Gem).to receive(:data_home) { "/a/b/c/.local/share" }
```

Comment:
I assume `data_home` will only be picked up if it exists? Or this PR is actually changing behavior and a credentials file in `~/.gem/credentials` will no longer be picked up?

Reply:
> I assume `data_home` will only be picked up if it exists?

Your expectation is correct. RG 3.2.0+ will pick `~/.gem/credentials` only if it exists.

Reply:
Right. So, the fact that other tests are leaving side effects is creating the false impression that the default has changed here. I'd like to try fix that.

Reply:
Same as https://github.com/rubygems/rubygems/pull/3545#discussion_r417241889

I felt your request as test structure. It's helpful for me if you submit the code about this. Thanks.